teapots/common/ndk_helper/JNIHelper.h

```diff
@@ -151,23 +235,31 @@ class JNIHelper {
   std::string GetExternalFilesDir();
 
   /*
+   * Retrieve string resource with a given name
+   * arguments:
+   *  in: resourceName, name of string resource to retrieve
+   * return: string resource value, returns "" when there is no string resource with given name
+   */
+  std::string GetStringResource(const std::string& resourceName);
```

Comment:
same as above: as long as you use code like:
  std::string s = std::string(resource);
  return s;

probably do not need to change the return type as const std::string.